CMSgov/design-system
# [WNMGDS-3044] Adds custom playwright reporter to provide JSON summary of test result data

## Summary

- Adds a custom Playwright reporter that generates a JSON summary of test results.
- The JSON report includes statistics on the number of passing, failing, and skipped tests, as well as details of failing tests and skipped tests.
- The primary objective is to make test statistics more accessible on the `test-nightly` pipeline in Jenkins.
- This effort lays the groundwork for tracking test coverage of browser tests.

[Jira Ticket](https://jira.cms.gov/browse/WNMGDS-3044)

### Example JSON output
<img width="952" alt="Screenshot 2024-11-26 at 11 01 46 AM" src="https://github.com/user-attachments/assets/7ed9e096-44fe-4bc8-8c7c-be54bced93d6">

## How to test

- Pull down branch.
- Run `yarn test:browser` or `yarn test:browser:examples` or any other test browser commands.
- Inspect logs: verify that hierarchy paths for failing tests are logged correctly during the run.
- Confirm that the `test-report.json` file is generated in the expected directory: `tests/browser`

## Checklist
